h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -837,8 +837,6 @@ public void init(boolean expensive) {
       BetaConstraint bc = (_parms._beta_constraints != null)?new BetaConstraint(_parms._beta_constraints.get()):new BetaConstraint();
       if((bc.hasBounds() || bc.hasProximalPenalty()) && _parms._compute_p_values)
         error("_compute_p_values","P-values can not be computed for constrained problems");
-      if(bc.hasBounds())
-        _parms._early_stopping = false; // PUBDEV-4641: early stopping does not work correctly with non-negative option
```

Comment:
Instead of deleting this, please change to this:

if (bc.hasBounds() && _parms._early_stopping)
  warn("beta constraint and early_stopping", "if both are enabled may degrade model performance.");

Reply:
Thanks @wendycwong, I have added the change